module-4/backend/week_one.md

```diff
@@ -1,21 +1,79 @@
-13. What's a background worker? When would we want to use a background worker?
+
+Week One - Module 4 Recap
+
+Fork this respository. Answer the questions to the best of your ability.
+Try to answer them with limited amount of external research.
+These questions cover the majority of what we've learned this week.
+
+Note: When you're done, submit a PR.
+
+    What's the most useful thing you learned from completing the intermission week work?
+      basics of javascript
+
+    What are some tools to help debug JavaScript code?
+      node
+      debugger
+
+    What are some tools you need in order to unit test your JavaScript?
+      initialize npm
+      chai
+      mocha
+      assert or expect
+
+    What is the syntax for invoking a function?
+      () {console.log("Hi");};
+      or
+      derp () {console.log("Hi");};
+
+    What's the difference between == and === in JavaScript?
+      === deeply true  
+      == true
+    What's the difference between asynchronous and synchronous JavaScript?
+      asynchronous JS is javascript running jobs so that certain jobs that
+      are pre-marked are started then while they are being worked on are
+      sort of put to the back in a way and other jobs are started  
+      when the job finishes it comes back just like normal
+
+      synchronous javascript is code that, naturally runs as a single thread and
+      when a job that takes long to complete comes into the que the thread
+      waits for the job to complete before moving on.
+
+    What's a callback function and what are some reasons when we use/need callback functions?
+      callback function is somethings thats used to enact the asynchronous action.
+
+    What's the biggest difference between a promise and a callback?
+      callback used several times indicate when asynchronous call are made
+      promises are made just once
+
+    How do we setup a route when creating an API with Node and Express?
+        var express = require('express');
+        var app = express();
+        var bodyParser = require('body-parser')
+
+        app.use(bodyParser.json())
+        app.use(bodyParser.urlencoded({ extended: true }))
+
+        app.set('port', process.env.PORT || 3000);
+        app.locals.title = "Quantified Self"
+
+        app.locals.food = {
+        wow: "Banana"
+        };
+
+        app.get('/', (request, response)  => {
+        response.send('Welcome to Quantified Self.')
+        });
+
+    What's npm and what do we use it for?
+      npm is the packet manager for javascript. use it to create projects and install packages.
+
+Review
+
+    What's the MVC design pattern? Describe each part of MVC?
+      Model - holds the models that model a single instance of something in a database
+      View - holds any views the browser sees
+      Controller - holds all the controller logic
+    What is AJAX? What are some benefits of using AJAX?
+      AJAX is a way to organize a output of an api
```

Comment:
AJAX requests are a way of connecting with a server (via an http request) and changing the content on a page without refreshing the entire page